dotCMS/src/integration-test/java/com/dotmarketing/util/PageModeTest.java

```diff
@@ -51,6 +60,50 @@ private HttpServletRequest backEndRequest() {
         return request;
     }
 
+    /**
+     * Method to test: Test for {@link VelocityServlet#processPageMode(User, HttpServletRequest)}
+     * Given Scenario: Be user logged in should be NAVIGATE_EDIT_MODE
+     * ExpectedResult: Page mode should be NAVIGATE_EDIT_MODE
+     *
+     */
+    @Test
+    public void test_be_page_mode_should_be_NAVIGATE_EDIT_MODE() {
+
+        final HttpSession session        = new MockSession(UUIDGenerator.uuid());
+        final MockSessionRequest requestSession = new MockSessionRequest(backEndRequest());
+        final MockAttributeRequest request = new MockAttributeRequest(requestSession);
+        session.setAttribute(com.dotmarketing.util.WebKeys.PAGE_MODE_SESSION, PageMode.NAVIGATE_EDIT_MODE);
+        request.setAttribute(WebKeys.USER, backEndUser);
+        requestSession.setSession(session);
+        LoginMode.set(request, LoginMode.BE);
+        HttpServletRequestThreadLocal.INSTANCE.setRequest(request);
+        final PageMode pageMode = VelocityServlet.processPageMode(backEndUser, request);
+        Assert.assertEquals(LoginMode.BE, LoginMode.get(request));
+        Assert.assertEquals(PageMode.NAVIGATE_EDIT_MODE, pageMode);
+    }
+
+    /**
+     * Method to test: Test for {@link VelocityServlet#processPageMode(User, HttpServletRequest)}
+     * Given Scenario: Fe user logged in should be live
+     * ExpectedResult: Page mode should be live
+     *
+     */
+    @Test
+    public void test_fe_page_mode_should_be_LIVE() {
+
+        final HttpSession session        = new MockSession(UUIDGenerator.uuid());
+        final MockSessionRequest requestSession = new MockSessionRequest(frontEndRequest());
+        final MockAttributeRequest request = new MockAttributeRequest(requestSession);
+        session.setAttribute(com.dotmarketing.util.WebKeys.PAGE_MODE_SESSION, PageMode.NAVIGATE_EDIT_MODE);
+        request.setAttribute(WebKeys.USER, frontEndUser);
+        requestSession.setSession(session);
+        HttpServletRequestThreadLocal.INSTANCE.setRequest(request);
+        LoginMode.set(request, LoginMode.FE);
+        final PageMode pageMode = VelocityServlet.processPageMode(frontEndUser, frontEndRequest());
+        Assert.assertEquals(LoginMode.FE, LoginMode.get(request));
+        Assert.assertEquals(PageMode.LIVE, pageMode);
+    }
+
```

Comment:
Maybe you are missing some test here:

- When a user with FE and BE roles is logging in FE, the PageMode should be equals to LIVE
- When a user with FE and BE roles is logging in BE, the PageMode should be equals to NAVIGATE_EDIT_MODE